client/lib/checkout/index.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * Internal dependencies
+ */
+import { cartItems } from 'lib/cart-values';
+import { managePurchase } from 'me/purchases/paths';
+
+export function getExitCheckoutUrl( cart, siteSlug ) {
+	let url = '/plans/';
+
+	if ( cartItems.hasDomainRegistration( cart ) ) {
+		url = '/domains/add/';
+	} else if ( cartItems.hasDomainMapping( cart ) ) {
+		url = '/domains/add/mapping/';
+	} else if ( cartItems.hasProduct( cart, 'offsite_redirect' ) ) {
+		url = '/domains/add/site-redirect/';
+	} else if ( cartItems.hasProduct( cart, 'premium_theme' ) ) {
+		url = '/design/';
+	}
+
+	url = url + siteSlug;
+
+	if ( cartItems.hasRenewalItem( cart ) ) {
```

Comment:
Shouldn't we put this condition on the top? We would have just to return url from inside to avoid all previous checks that get override by this logic anyway.